x/stakeibc/keeper/unbonding_records.go

```diff
@@ -40,120 +40,144 @@ func (k Keeper) CreateEpochUnbondingRecord(ctx sdk.Context, epochNumber uint64)
-			k.Logger(ctx).Info(fmt.Sprintf("[SendHostZoneUnbondings] Sending unbondings, host zone: %s, epochUnbonding: %v", hostZone.ChainId, epochUnbonding))
-
+			k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId, "  %v%s included in total unbonding", hostZoneRecord.NativeTokenAmount, hostZoneRecord.Denom))
 		}
 	}
-	delegationAccount := hostZone.GetDelegationAccount()
-	if delegationAccount == nil || delegationAccount.GetAddress() == "" {
-		errMsg := fmt.Sprintf("Zone %s is missing a delegation address!", hostZone.ChainId)
-		k.Logger(ctx).Error(errMsg)
-		return nil, sdk.ZeroInt(), nil, nil, sdkerrors.Wrap(types.ErrHostZoneICAAccountNotFound, errMsg)
-	}
-	validators := hostZone.GetValidators()
-	if totalAmtToUnbond.IsZero() {
+	k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId, "Total unbonded amount: %v%s", totalAmountToUnbond, hostZone.HostDenom))
+
+	// If there's nothing to unbond, return and move on to the next host zone
+	if totalAmountToUnbond.IsZero() {
 		return nil, sdk.ZeroInt(), nil, nil, nil
 	}
-	// we distribute the unbonding based on our target weights
-	newUnbondingToValidator, err := k.GetTargetValAmtsForHostZone(ctx, hostZone, totalAmtToUnbond)
+
+	// Determine the desired unbonding amount for each validator based based on our target weights
+	targetUnbondingsByValidator, err := k.GetTargetValAmtsForHostZone(ctx, hostZone, totalAmountToUnbond)
 	if err != nil {
-		errMsg := fmt.Sprintf("Error getting target val amts for host zone %s %d: %s", hostZone.ChainId, totalAmtToUnbond, err)
+		errMsg := fmt.Sprintf("Error getting target val amts for host zone %s %v: %s", hostZone.ChainId, totalAmountToUnbond, err)
 		k.Logger(ctx).Error(errMsg)
 		return nil, sdk.ZeroInt(), nil, nil, sdkerrors.Wrap(types.ErrNoValidatorAmts, errMsg)
 	}
-	valAddrToUnbondAmt := make(map[string]sdk.Int)
-	overflowAmt := sdk.ZeroInt()
-	for _, validator := range validators {
-		valAddr := validator.GetAddress()
-		valUnbondAmt := newUnbondingToValidator[valAddr]
-		k.Logger(ctx).Info(fmt.Sprintf("Validator %s - Weight: %d, Expected Unbond Amount: %d, Current Delegations: %d", valAddr, validator.Weight, valUnbondAmt, validator.DelegationAmt))
-
-		if valUnbondAmt.GT(validator.DelegationAmt) { // if we don't have enough assets to unbond
-			overflowAmt = overflowAmt.Add(valUnbondAmt).Sub(validator.DelegationAmt)
-			valUnbondAmt = validator.DelegationAmt
-		}
-		if err != nil {
-			errMsg := fmt.Sprintf("Error casting validator staked amount %d: %s", validator.DelegationAmt, err.Error())
-			k.Logger(ctx).Error(errMsg)
-			return nil, sdk.ZeroInt(), nil, nil, sdkerrors.Wrap(types.ErrIntCast, errMsg)
+
+	// Check if each validator has enough current delegations to cover the target unbonded amount
+	// If it doesn't have enough, update the target to equal their total delegations and record the overflow amount
+	actualUnbondingsByValidator := make(map[string]sdk.Int)
```

Comment:
the name `actualUnbondingsByValidator` is a bit unclear to me, maybe just `unbondingsByValidator`? Open to other names here as well!

Reply:
yeah I was trying to separate out the logical steps of 1) first getting target unbondings, and then 2) adjusting them in the case of overflow

I'm good with changing it to unbondingsByValidator, but might make sense to rename targetUnbondingsByValidator to that (maybe even just share the same variable)? Also if you can think of a better word that differentiates step 2 that'd also work

Reply:
Ah I see - I think it's already implied, if we start with `target...` and then have `unbondings...`. `final...` could work as well!

Reply:
I like final!